akka-actor/src/main/scala/akka/actor/FSM.scala

```diff
@@ -784,7 +784,7 @@ trait LoggingFSM[S, D] extends FSM[S, D] { this: Actor ⇒
         case a: ActorRef          ⇒ a.toString
         case _                    ⇒ "unknown"
       }
-      log.debug("processing " + event + " from " + srcstr)
+      log.debug("processing " + event + " from " + srcstr + " in state " + stateName)
```

Comment:
Please change this to `debug("processing {} from {} in state {}", event, srcstr, stateName)` so that we do not construct this string even when logging is not enabled. Same below.